app/src/main/java/tech/ula/utils/ServerUtility.kt

```diff
@@ -57,13 +55,19 @@ class ServerUtility(
     }
 
     /**
-     * Execute the boot script, should be called only when the session has the startOnBoot flag
-     * Users can use this to add their own custom commands to this script.
+     * Execute the user-defined startCommand
+     * Creates a new script based on userland_profile.sh and adds the user-defined command to the script
      */
-    private fun runBootScript(session: Session) {
+    fun executeStartCommand(session: Session) {
+        if (session.startCommand.isEmpty())
+            return
         val filesystemDirName = session.filesystemId.toString()
-        // TODO create new session parameter and turn this into a user-defined command
         val command = "/support/autostart.sh"
```

Comment:
I think at this point, `command` is a slightly disingenuous name for this variable. I think it might make the most sense to just remove it and change `startScript` to include that part of the path.